src/mame/drivers/gfamily.cpp

```diff
@@ -0,0 +1,120 @@
+// license:BSD-3-Clause
+// copyright-holders:
+/*
+Games Family P4-4P (Pentium 4 - 4 Players)
+
+PC with Chinese Windows 2000 Pro and several emulators, including:
+ -MAME v0.78 (Dec 30 2003)
+ -MAME v0.96u3 (May 25 2005)
+ -MAME v0.106 (May 16 2006)
+ -Nebula 2.1.5
+ -FB Alpha v0.2.94.98
+ -ZiNc 1.1
+
+
+PC motherboard plus an additional PCB for JAMMA, inputs and basic config.
+
+PC board:
+CPU: Intel Celeron 1,7GHz / 128kb / 400MHz SL6SC
+RAM: 256MB-DDR400
+PCB: 04/082006-SiS-651-6A6IXRMAC-00, Realtec ALC655 audio, Realtec RTL8201BL Ethernet (25.000 MHz xtal), Winbond W83194BG-648 (14.31818 xtal)
+BIOS: 686 AMIBIOS (c) 2006 AY36 8897
+
+I/O board:
+AT89C2051 + Microchip CF745 + 2 x Microchip PIC12F508 + Altera Max CPLD
+
+*/
+
+#include "emu.h"
+#include "cpu/i386/i386.h"
+#include "screen.h"
+
+namespace {
+
+class gfamily_state : public driver_device
+{
+public:
+	gfamily_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu")
+	{ }
+
+	void gfamily(machine_config &config);
+
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+	virtual void video_start() override;
+
+private:
+	required_device<cpu_device> m_maincpu;
+
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+	void gfamily_map(address_map &map);
+};
+
+void gfamily_state::video_start()
+{
+}
+
+uint32_t gfamily_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+{
+	return 0;
+}
+
+void gfamily_state::gfamily_map(address_map &map)
+{
+}
+
+static INPUT_PORTS_START( gfamily )
+INPUT_PORTS_END
+
+
+void gfamily_state::machine_start()
+{
+}
+
+void gfamily_state::machine_reset()
+{
+}
+
+void gfamily_state::gfamily(machine_config &config)
+{
+	// Basic machine hardware
+	PENTIUM4(config, m_maincpu, 1700000000); // Actually an Intel Celeron SL6SC 1,7GHz
```

Comment:
Please use digit grouping in large numbers to it’s easier to see the order of magnitude.